src/frontend/src/flows/addDevice/manage/addRemoteDevice.ts

```diff
@@ -0,0 +1,55 @@
+import { AuthenticatedConnection } from "../../../utils/iiConnection";
+import { pollForTentativeDevice } from "./pollForTentativeDevice";
+import { displayError } from "../../../components/displayError";
+import { verifyTentativeDevice } from "./verifyTentativeDevice";
+import { withLoader } from "../../../components/loader";
+
+// The flow for adding a remote (i.e. other browser, non-hardware) device
+export const addRemoteDevice = async ({
+  userNumber,
+  connection,
+}: {
+  userNumber: bigint;
+  connection: AuthenticatedConnection;
+}): Promise<void> => {
+  // Enter registration mode and get info about the anchor
+  const [timestamp, anchorInfo] = await withLoader(() =>
+    Promise.all([
+      connection.enterDeviceRegistrationMode(),
+      connection.getAnchorInfo(),
+    ])
+  );
+
+  let tentativeDevice = anchorInfo.device_registration[0]?.tentative_device[0];
+  if (tentativeDevice === undefined) {
+    // If no device was tentatively added yet, poll until one is added
+    const result = await pollForTentativeDevice(
+      userNumber,
+      connection,
+      timestamp
+    );
+
+    if (result === "timeout") {
+      // On timeout, notify the user
+      await displayError({
+        title: "Timeout Reached",
+        message:
+          'The timeout has been reached. For security reasons the "add device" process has been aborted.',
+        primaryButton: "Ok",
+      });
+      return;
```

Comment:
can result be here something else that timeout or that's an edge case? can you maybe add a comment next to this return to explains what happens if we land here?

Reply:
The result can be a timeout, a cancellation or device information. For the two first cases (timeout, cancel) there's not much we can do. I'm adding a comment!